runtimes/CMakeLists.txt

```diff
@@ -91,6 +91,51 @@ include(CheckLibraryExists)
 include(LLVMCheckCompilerLinkerFlag)
 include(CheckCCompilerFlag)
 include(CheckCXXCompilerFlag)
+include(ExtendPath)
+
+
+# CMake 3.24 is the first version of CMake that directly recognizes Flang.
+# LLVM's requirement is only CMake 3.20, teach CMake 3.20-3.23 how to use Flang, if used.
+if (CMAKE_VERSION VERSION_LESS "3.24")
```

Comment:
The flang support in CMake wasn't really complete/fully functional until 3.28 so I think this should be bumped up to that. Unless you've specifically tested with 3.24 and it doesn't hit the edge cases we didn't fix until later?

Reply:
I think I tests 3.24 already. I boiled down to `CMAKE_Fortran_PREPROCESS_SOURCE` not being defined. I will test again before landing.